File: src/saveHtml.ts
import * as path from "node:path";
import * as fs from "node:fs";

/**
 * HTMLをファイルに保存する
 * @param html 保存するHTML文字列
 * @param filePath 保存先のファイルパス
 */
export const saveHTMLToFile = (html: string, filePath: string) => {
  const dir = path.dirname(filePath);

  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, {recursive: true});
  }

  fs.writeFileSync(filePath, html, 'utf8');
}

File: src/save_html.py
def save_html_to_file(html: str, filepath: str) -> None:
    """
    HTMLをファイルに保存する
    
    Args:
        html (str): 保存するHTML文字列
        filepath (str): 保存先のファイルパス
    """
    try:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(html)
    except IOError as e:
        raise Exception(f"ファイルの保存に失敗しました: {str(e)}") 
